source/languageHandler.py

```diff
@@ -79,20 +115,90 @@ def getLanguageDescription(language):
 		desc=buf.value
 	if not desc:
 		#Some hard-coded descriptions where we know the language fails on various configurations.
-		desc={
-			# Translators: The name of a language supported by NVDA.
-			"an":pgettext("languageName","Aragonese"),
-			# Translators: The name of a language supported by NVDA.
-			"ckb":pgettext("languageName","Central Kurdish"),
-			# Translators: The name of a language supported by NVDA.
-			"kmr":pgettext("languageName","Northern Kurdish"),
-			# Translators: The name of a language supported by NVDA.
-			"my":pgettext("languageName","Burmese"),
-			# Translators: The name of a language supported by NVDA.
-			"so":pgettext("languageName","Somali"),
-		}.get(language,None)
+		# Imported lazily since langs description are translatable
+		# and `languageHandler` is responsible for setting the translation.
+		import localesData
+		desc = localesData.LANG_NAMES_TO_LOCALIZED_DESCS.get(language, None)
 	return desc
 
+
+def englishLanguageNameFromNVDALocale(localeName: str) -> Optional[str]:
+	"""Returns either English name of the given language  using `GetLocaleInfoEx` or None
+	if the given locale is not known to Windows."""
+	localeName = normalizeLocaleForWin32(localeName)
+	buffLength = winKernel.kernel32.GetLocaleInfoEx(localeName, LOCALE_SENGLISHLANGUAGENAME, None, 0)
+	if buffLength:
+		buf = ctypes.create_unicode_buffer(buffLength)
+		winKernel.kernel32.GetLocaleInfoEx(localeName, LOCALE_SENGLISHLANGUAGENAME, buf, buffLength)
+		langName = buf.value
+		if "Unknown" in langName:
+			# Windows 10 returns 'Unknown' for locales not known to Windows
+			# even though documentation states that in case of an unknown locale 0 is returned.
+			return None
+		try:
+			langName.encode("ascii")
+			return langName
+		except UnicodeEncodeError:
+			# The language name cannot be encoded in ASCII which unfortunately means we wonn't be able
+			# to set Python's locale to it (Python issue 26024).
+			# this has been observed for Norwegian
+			# (language name as returned from Windows is 'Norwegian Bokmål').
+			# Thankfully keeping just the ASCII part of the string yields the desired result.
+			partsList = []
+			for part in langName.split():
+				try:
+					part.encode("ascii")
+					partsList.append(part)
+				except UnicodeEncodeError:
+					continue
+			return " ".join(partsList)
+	return None

```

Comment:
Could you add a test to confirm None is returned in this case?

Reply:
This is being tested by `test_noLangNameFromUnknownLocale` as explained above.